### ISSUE TITLE
skindesigner not working with latest commit

### ISSUE THREAD

**Issue by jupe76** [pesintta/vdr-plugin-vaapidevice#67]
After updating to the latest set of commits, OSD is not longer shown with skindesigner.
VDR-native OSD (i.e. LCARS) is ok.

Digging in the log, this may give a hint.
Mar 04 20:07:36 vdrtwo vdr[6575]: [6575] ERROR: cOsd::SetAreas returned 5 (wrong alignment)
Mar 04 20:07:36 vdrtwo vdr[6575]: [6575] skindesigner: Error initiating displaychannel view - aborting

**Comment by pesintta** [comment]
I tried with vdr-2.3.8, skindesigner (latest master) and vaapidevice (latest master) and could not reproduce such behavior.

Have you tried running vaapidevice in fullscreen (-f switch)? It looks like some osd area might be supplied wrong but I don't yet know how to track that. It would be helpful if you could add more debug logging to vdr/skindesigner

More specifically if you could log x, y and width and height in skindesigners cSdOsd::CreateOsd method.

**Comment by jupe76** [comment]
I'm also running vdr 2.3.8, latest skindesigner master.
Parameters for vaapidevice are -a hw:0,3 -p hw:0,3 -D -f

I'll look after the needed debug informations ...

**Comment by jupe76** [comment]
Ok, so with 6372704835b62bee882feed92686edc75e70b55f I'm getting:
Mar 04 22:04:37 vdrtwo vdr[14474]: [14474] CreateOsd: x 0, y 0, width 1920, height 1080

With master I'm getting:
Mar 04 22:06:48 vdrtwo vdr[14957]: [14957] ERROR: unknown config parameter: vaapidevice.Osd.Height = 0
Mar 04 22:06:48 vdrtwo vdr[14957]: [14957] ERROR: unknown config parameter: vaapidevice.Osd.Width = 0
Mar 04 22:06:49 vdrtwo vdr[14957]: [14957] CreateOsd: x 0, y 0, width 0, height 0
Mar 04 22:06:49 vdrtwo vdr[14957]: [14957] ERROR: cOsd::SetAreas returned 5 (wrong alignment)

**Comment by Space2Man** [comment]
Hi,

I can confirm the issue ... running latest vaapidevice + latest skindesigner ... using shady-KISS skin ... had to revert because of WAF issue ... I will check for downtime window :)

**Comment by pesintta** [comment]
I pushed a commit https://github.com/pesintta/vdr-plugin-vaapidevice/commit/1198cfd43ba311b7349b723bc1ad4ebb675f3227 to master that might have an effect on this. Can you test that?

**Comment by jupe76** [comment]
Hi, thanks, that did the trick.
OSD is working again.

Only drawback is that now the log is sometimes flooded with:
Mar 04 22:59:12 vdrtwo vdr[18174]: [18174] VAAPI-ERROR: video: VideoGetOsdSize: osd/window size not set yet

**Comment by Space2Man** [comment]
Hi,

I see different error messages ... I get (with latest commit):

```
Mar 05 00:11:24 [vdr] [18025] VAAPI: Device ready
Mar 05 00:11:24 [vdr] [18067] VAAPI-ERROR: video/vaapi: no osd subpicture yet
Mar 05 00:11:24 [vdr] [18067] VAAPI-ERROR: video: display buffer empty, duping frame (1/0) 0
Mar 05 00:11:24 [vdr] [18067] VAAPI-ERROR: video/vaapi: no osd subpicture yet
Mar 05 00:11:24 [vdr] [18067] VAAPI-ERROR: video: display buffer empty, duping frame (2/0) 0
Mar 05 00:11:24 [vdr] [18067] VAAPI-ERROR: video/vaapi: no osd subpicture yet

```
And after some time:

```
Mar 05 00:11:25 [vdr] [18067] VAAPI-ERROR: video/vaapi: no osd subpicture yet
Mar 05 00:11:25 [vdr] [18067] VAAPI-ERROR: video: display buffer empty, duping frame (585/0) 6
Mar 05 00:11:25 [vdr] [18067] VAAPI-ERROR: video/vaapi: can't destroy 8 surfaces
Mar 05 00:11:25 [vdr] [18067] VAAPI-ERROR: video/vaapi: can't destroy postproc context!
Mar 05 00:11:25 [vdr] [18067] VAAPI-ERROR: video/vaapi: can't destroy config!
Mar 05 00:11:25 [vdr] [18067] VAAPI: video/vaapi: supports video processing
Mar 05 00:11:25 [vdr] [18067] VAAPI-ERROR: video/vaapi: no osd subpicture yet
Mar 05 00:11:25 [vdr] [18067] VAAPI: video/vaapi: noise reduction supported
```
Best regards,

   Space

**Comment by jupe76** [comment]
closed with 1198cfd43ba311b7349b723bc1ad4ebb675f3227
